Feature/todak 174/public diary find

kchsoft opened the pull request:
<details>
<summary> PR 체크리스트</summary>
  
- [x] 🔀 PR 제목의 형식을 잘 작성했나요? e.g. #1 [feat] 소셜 로그인 구현
- [x] 💭 이슈는 등록했나요?
- [x] 🏷️ 라벨은 등록했나요?
- [x ] 🙇‍♂️ 리뷰어를 지정했나요? (페어가 존재할 시)
</details>

## 변경사항

- 공개 일기를 조회하는 무한 스크롤 API를 추가하였습니다.
- 10/31 기획 변경에 맞추어 s3 pre-signed를 할 수 있는 service를 추가 하였고, DB는 하나의 URL을, Spring은 4개의 웹툰 URL을 전달하는 구조로 구현하였습니다. 그러나 실질적인 pre-signed url 구현은 추후에 진행될 예정입니다. 

## 리뷰 받고 싶은 내용

- 메서드 이름 부터 객체 이름 짓는 것 + ```PublicDiaryServcie```의 메서드 및 객체가 많아져 점점 보기 복잡해 지는 것 같습니다. 객체지향적인 좋은 코드 방향이 있다면 알려주시면 감사하겠습니다!

```PublicDiaryViewDetailResponse``` : 무한 스크롤에 대한 최종적인 응답 객체 입니다.
```PublicDiaryViewProjection```  :  공개 일기 ID를 바탕으로 DB에서 가져온 ```공개 일기 정보``` 입니다. (Reaction 없음)
```PublicDiaryView``` : 바로 위의 ```Projection```에서 pre-signed를 처리한 객체 입니다.  웹툰 url의 경우 List의 형태로 4개의 URL을 가집니다.
```PublicDiaryViewDetail``` : ```PublicDiaryView``` + ```Detail``` (= [ Reaction 갯수 / 사용자가 어떤 Reaction을 눌렀는지 ] ) 에 대한 정보를 추가로 담고 있는 객체입니다.
``` DiaryReactionCountProjection``` : DB로 부터 일기장에 몇 개의 Reaction이 달려있는지 정보를 가져온 객체입니다.


## 테스트 및 결과
- 로컬 테스트 확인 완료.

close #56 

## 관련 스크린샷 및 참고자료 (Optional)

---
# 2024/11/01 코드 리팩토링
- 아래와 같이 객체 이름 및 메서드 구조 등을 변경하였습니다.
![image](https://github.com/user-attachments/assets/2d9d9ecb-baac-47b0-b9ec-31633bec80a0)

Servie 코드의 책임을 크게 3가지로 분리하였습니다.
1. 일기 정보 조회
2. pre-sigend url 설정
3. 반응 조회 및 응답 생성

객체 이름은 아래와 같이 변경됩니다.
```PublicDiaryViewDetailResponse``` -> ```PublicDiaryPaginationlResponse```
```PublicDiaryViewProjection``` -> ```PublicDiaryContentOnlyProjection```
```PublicDiaryView``` -> pre-signed url 설정 과정이 단순해짐에 따라 객체가 삭제 되었습니다.
```PublicDiaryViewDetail``` -> ```PublicDiary``` 
``` DiaryReactionCountProjection``` -> 동일하게 유지합니다.


- 기존에는 ```Projectoin```은 웹툰 url을 ```String``` 형태로 가지고 있었지만, 현재는 DB에서 얻어온 url을 생성자에서 List 형태로 변환하여 저장합니다. 이를 통해 pre-signed url을 설정하는 과정을 단순화 하였습니다.

kchsoft committed:
feat: 공개 일기 조회 API 관련 객체 생성 TODAK-174

kchsoft committed:
feat: 공개 일기 조회 API 비즈니스 로직 작성 TODAK-174

kchsoft committed:
feat: 공개 일기 조회 API 로그 작성 TODAK-174

kchsoft committed:
feat: 공개 일기 조회 API swagger 적용 TODAK-174

- 불필요한 projection 삭제

kchsoft committed:
feat: s3 url pre-signed 객체 생성 TODAK-174

kchsoft committed:
test: 공개 일기 조회 API 테스트 작성 TODAK-174

- response 객체 초기화 오류 수정

kchsoft committed:
feat: 공개 일기 조회 API swagger 추가 적용 TODAK-174

Urchinode commented on src/main/java/com/heartsave/todaktodak_api/diary/dto/PublicDiaryView.java:
```diff
@@ -0,0 +1,51 @@
+import java.util.List;
+import lombok.Getter;
+
+@Getter
```
이 부분도 `@Schema` description으로 어떤 객체인지 설명이 있으면
객체 역할을 쉽게 알 수 있을 것 같습니다 😃

kchsoft committed:
Merge branch 'develop' into feature/TODAK-174/public-diary-find

Urchinode commented on src/main/java/com/heartsave/todaktodak_api/diary/repository/PublicDiaryRepository.java:
```diff
@@ -1,6 +1,37 @@
+        JOIN pd.diaryEntity d
+        JOIN pd.memberEntity m
+        WHERE pd.id < :publicDiaryId
+        ORDER BY pd.id DESC
```
이 메서드는 `Pageable`로 인해 `LIMIT 5`가 적용되어 sql이 생성되나요?
아니면 `Pageable`이 전체 데이터에서 5개만큼 추출해내는지 궁금합니다.

kchsoft committed:
refactor: 공개 일기 조회의 Projection 필드 변경 TODAK-174

- String -> List<String> 으로 변경하여, s3 url을 pre-signed url로 변경하여 projection에 반영하도록 변경

kchsoft committed:
refactor: 공개 일기 조회 API 관련 객체 이름 변경 TODAK-174

- View -> PublicDiaryContentOnly
- ViewDetail -> PublicDiary
- ViewDetailResponse -> PaginationResponse

kchsoft committed:
refactor: 공개 일기 조회 API 코드 개선 TODAK-174

- 메서드 분리
- Projcetion의 url 필드를 List로 변화
- PublicDiary 예외 처리

kchsoft committed:
refactor: 공개 일기 조회 API 메서드명 변경 TODAK-174

kchsoft committed:
refactor: 공개 일기 예외 발생 http status 변경 TODAK-174

kchsoft committed:
refactor: view, detail 이름 변경 TODAK-174
